Upgrade to arrow 57

mjgarton opened the pull request:
Upgrade the version of arrow from 56 to 57.

mjgarton committed:
Upgrade to arrow 57

Upgrade the version of arrow from 56 to 57.

mlafeldt commented:
Thanks for the PR.

While this update may seem benign, it would also require us to bump the MSRV to `1.85.1` due to https://github.com/apache/arrow-rs/blob/57.0.0/Cargo.toml#L85

However, [we committed](https://github.com/duckdb/duckdb-rs?tab=readme-ov-file#rust-version-compatibility) to not change that version in patch releases (1.4.3 would be [next](https://duckdb.org/release_calendar)).

Is there anything in particular you need from Arrow v57?

mlafeldt committed:
Update Cargo.lock

mlafeldt committed:
Bump MSRV to what Arrow 57 requires

mlafeldt committed:
Fix test for Arrow 57 stricter FixedSizeListArray validation

Arrow 57.1.0+ requires values buffer length to be a multiple of the fixed size.
